SingularityExecutor/src/main/java/com/hubspot/singularity/executor/shells/SingularityExecutorShellCommandRunner.java

```diff
@@ -119,14 +124,39 @@ public boolean apply(SingularityExecutorShellCommandDescriptor input) {
 
     final SingularityExecutorShellCommandDescriptor shellCommandDescriptor = matchingShellCommandDescriptor.get();
 
-    List<String> command = new ArrayList<>(shellCommandDescriptor.getCommand());
+    List<String> command = new ArrayList<>();
+
+    if (shellCommandDescriptor.isSwitchUser()) {
+      String switchUserCommand = String.format(executorConfiguration.getSwitchUserCommandFormat(), taskProcess.getTask().getExecutorData().getUser().or(executorConfiguration.getDefaultRunAsUser()));
+      command.addAll(Arrays.asList(switchUserCommand.split("\\s+")));
```

Comment:
seems a little janky, might be better to explicitly define what the switch user command for shell commands is